### PR TITLE
Fix UI not updating and change free command syntax

### LINKED ISSUE #131: Free command: change command syntax
Why do you ever need to specify which driver to free a task from??? Just free the task!!

### LINKED ISSUE #148: Ui - Driver/Task card not updating
Upon `assign` command, the changes should reflect in the Task and Driver cards, but it didn't.

This is because the backing data structure of `TaskList` and `DriverManager` only watches changes to the elements itself, i.e. adding/deleting/replacing. Since `Driver` is an attribute of `Task` and `Schedule` is an attribute of `Driver`, updating Driver and Schedule will not trigger the refresh of their container classes.

### PR DESCRIPTION
**Pull request by tysg**
Closes #131 
Closes #148

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AssignCommand.java [hunk 1]
```diff
@@ -67,12 +70,18 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Driver driver = model.getDriver(driverId);
         Task task = model.getTask(taskId);
+        if (task.getStatus() != TaskStatus.INCOMPLETE || task.getDriver().isPresent()
+                || task.getEventTime().isPresent()) {
+            throw new CommandException(MESSAGE_ALREADY_ASSIGNED);
+        }
```
Comment (junnbang) [inline src/main/java/seedu/address/logic/commands/AssignCommand.java]: having the task.getDriver().isPresent() || task.getEventTime().isPresent() seems like saying there's a possibility that only 1 of the field is populated which shouldn't be possible. I suggest creating a boolean method in task to check both at the same time.

**Review by junnbang — APPROVED** [review]
loooks fine!

File: src/main/java/seedu/address/logic/commands/AssignCommand.java (hunk 1 above)
Comment (tysng) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java]:
```
task.getStatus() != TaskStatus.INCOMPLETE && task.getDriver().isPresent() && task.getEventTime().isPresent()
```
Should be this right?

Comment (tysng) [reply src/main/java/seedu/address/logic/commands/AssignCommand.java]: It's more like defensive programming. If a task is not ongoing, then the driver and eventTime should be absent, only if the other classes are implemented correctly. Here I'm failing the command if any of those predicate is true.

**Pull request merged**